scionlab/tests/test_user_as_views.py

```diff
@@ -39,82 +43,71 @@ def _create_ases_for_testuser(num):
-        UserAS.objects.create(
+        ap = AttachmentPoint.objects.first()
+        user_as = UserAS.objects.create(
             owner=user,
-            attachment_point=AttachmentPoint.objects.first(),
-            public_ip=_test_ip,
-            public_port=_test_start_port + i,
+            installation_type=UserAS.VM,
+            isd=ap.AS.isd,
             label="Testuser's AS number %i" % (i + 1),
-            installation_type=UserAS.VM
         )
+        ap_conf = AttachmentConf(ap,
+                                 _test_ip, _test_start_port + i,
+                                 bind_ip=None, bind_port=None,
+                                 use_vpn=False)
+        user_as.update_attachments([ap_conf])
 
 
 class UserASFormTests(TestCase):
     fixtures = ['testdata']
 
-    # Valid form parameters
-    valid_form_params = [
-        param(attachment_point="1", use_vpn=True),
-        param(attachment_point="1", use_vpn=True, installation_type='VM'),
-        param(attachment_point="1", use_vpn=True, installation_type='PKG'),
-        param(attachment_point="1", use_vpn=True, installation_type='SRC'),
-        param(attachment_point="1", use_vpn=True, label="This is a label"),
-        param(attachment_point="1", use_vpn=True, public_port=54321),
-        param(attachment_point="1", public_ip=_test_ip),
-        param(attachment_point="1", public_ip=_test_ip, public_port=54321),
-        # Ok to provide redundant IP:
-        param(attachment_point="1", use_vpn=True, public_ip=_test_ip),
-        param(attachment_point="2", public_ip=_test_ip),
-        param(attachment_point="1", use_vpn=True, public_ip=_test_ipv6),
-    ]
-    # Invalid form parameters
-    invalid_form_params = [
-        # Needs either VPN or public IP
-        param(attachment_point="1", use_vpn=False, public_ip=''),
-        # AP 2 does not have VPN
-        param(attachment_point="2", use_vpn=True),
-        # invalid IP
-        param(attachment_point="1", use_vpn=False, public_ip='aa'),
-        # localhost is not allowed
-        param(attachment_point="1", use_vpn=False, public_ip='127.0.0.1'),
-        # the attachment point doesn't support IPv6
-        param(attachment_point="1", use_vpn=False, public_ip=_test_ipv6),
-    ]
+    useras_initial = {
+        "user-as-label": "",
+        "user-as-installation_type": "VM"
+    }
+    attachment_initial = {
+        "active": "on", "use_vpn": "",
+        "attachment_point": "", "id": "",
+        "public_ip": "", "public_port": "50000",
+        "bind_ip": "", "bind_port": ""
+    }
+    # Valid form cases
+    valid_cases_file_path = path.join(_FORM_CASES_FOLDER, 'test_user_as_views_valid_forms.yaml')
+    valid_form_cases = yaml.safe_load(open(valid_cases_file_path).read())
+    # Invalid form cases
+    invalid_cases_file_path = path.join(_FORM_CASES_FOLDER, 'test_user_as_views_invalid_forms.yaml')
+    invalid_form_cases = yaml.safe_load(open(invalid_cases_file_path).read())
```

Comment:
As far as I can see, ths data is always passed through `_get_form_fields`. Could you apply this transformation here directly to avoid a bunch of boilerplate code?

Reply:
Cool, never heard of `pathlib`